core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -778,19 +829,27 @@ object Doc {
          * Making the recursion in the second parameter of the union fixes this.
          * This is the motivation for defer/LazyDoc.
          */
-        val (flaty, changedy) = y.flattenBoolean
-        val (flatx, changedx) = x.flattenBoolean
```

Comment:
I'd love to not change this code... couldn't we `flatx.getOrElse(x)` or something up here and use the same code?

Reply:
How would this proposal work?  We need different cases in the body depending on whether trying to flatten `x` for example returned `None` or not, don't we?  I would be happy to move the `None` cases to a different block if you'd like to preserve the existing handling of `Some` cases.  Wdyt?

Reply:
This may work since we're abandoning hope of making `fill` behave efficiently.  Working on it.

Reply:
cool